src/components/10-atoms/icon/index.js

```diff
@@ -67,21 +68,20 @@ class AXAIcon extends LitElement {
     const { icon } = this;
 
     if (/\.svg/.test(icon)) {
-      const req = new XMLHttpRequest();
-      req.open('GET', icon);
-      req.onreadystatechange = () => {
-        if (req.readyState === 4 && req.status === 200) {
-          this.size = 'auto';
-          this.shadowRoot.innerHTML = req.response;
-        }
-      };
-
-      req.send();
+      xhrCall(icon).then(result => {
+        this.size = 'auto';
+        this._loadedSvg = result;
+      });
+    } else if (/<svg/.test(icon)) {
+      this.size = 'auto';
+      this._loadedSvg = icon;
+    } else {
+      this._loadedSvg = AXAIcon.iconsMapping[icon] || '';
```

Comment:
Doesn't this create a boolean expression instead of a string?

Reply:
No, read it that way:

If `AXAIcon.iconsMapping[icon] ` is falsy, take the second part of OR. Which is empty string.